.travis.yml

```diff
@@ -6,8 +6,17 @@ before_script:
   - "npm install"
   - "npm run build"
   - "java -jar selenium-server-standalone-2.53.0.jar -Dphantomjs.binary.path=./node_modules/phantomjs-prebuilt/bin/phantomjs &"
-  - sleep 3
   - npm start &
-  - sleep 5
+  - "chmod 750 waiting-for-start.sh"
+  - "./waiting-for-start.sh"
+  - "STANDALONE=$(pgrep -n -f java)"
+  - "NODE=$(pgrep -n -f node)"
+  - "NPM=$(pgrep -n -f npm)"
```

Comment:
I don't think you need to care npm. `npm start` runs `node start` and waits till node stops. Once you kill node, npm will end immediately.